test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -135,6 +135,40 @@ class test_ucp_sockaddr : public ucp_test {
         return skip;
     }
 
+    void adjust_if_loopback() {
+        const char *ipv4_addr = "127.0.0.1";
+        const char *ipv6_addr = "::1";
+        const uint16_t port   = 0;
+        struct sockaddr_in lo_ipv4;
+        struct sockaddr_in6 lo_ipv6;
+        ucs_status_t status;
+
+        lo_ipv4.sin_family  = AF_INET;
+        lo_ipv4.sin_port    = htons(port);
+        lo_ipv6.sin6_family = AF_INET6;
+        lo_ipv6.sin6_port   = htons(port);
+
+        status = ucs_sock_ipstr_to_sockaddr(ipv4_addr,
+                                            (struct sockaddr_storage *)&lo_ipv4);
+        ASSERT_UCS_OK(status);
+
+        status = ucs_sock_ipstr_to_sockaddr(ipv6_addr,
+                                            (struct sockaddr_storage *)&lo_ipv6);
+        ASSERT_UCS_OK(status);
+
+        if (!ucs_sockaddr_cmp(m_test_addr.get_sock_addr_ptr(),
+                              (struct sockaddr *)&lo_ipv4, &status) ||
```

Comment:
will it be static checker issue to expect that `status == UCS_OK`, but setting it for ipv4 comparison and then for ipv6 comparison? i.e. if ipv4 fails with `UCS_ERR_INVALID_PARAM` and then ipv6 comparison is `UCS_OK`, the assertion on line 163 will be successful, but it is unexpected

Reply:
@dmitrygx i'll split to 2 statuses then.

Reply:
> @dmitrygx i'll split to 2 statuses then.

@alinask I think it is better
thanks

Reply:
thanks! 👍